### PR TITLE
Fix ed25519 key type in ca_util

### PR DESCRIPTION
**Pull request by rculpepper**
This PR is to address the error described in issue #26089 with ed25519 keys

### CODE REVIEW

**Review by victorr — APPROVED** [review]
👍 

Is there maybe a unit test we can augment to cover this case?

**Comment by rculpepper** [comment]
> Also what are your thoughts on just adding both the pointer and non-pointer types to all the case statements?

I don't have a problem with it, but I don't know how useful it would be. I don't see anywhere that we expect an `ed25519` key to be a pointer (except the case here)

File: changelog/27093.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+pki: Fix error in cross-signing using ed25519 keys
```
Comment (stevendpclark) [inline changelog/27093.txt:2]:
```suggestion
secrets/pki: Fix error in cross-signing using ed25519 keys
```

**Review by stevendpclark — APPROVED** [review]

**Comment by stevendpclark** [comment]
> I don't have a problem with it, but I don't know how useful it would be. I don't see anywhere that we expect an `ed25519` key to be a pointer (except the case here)

Fair, lets not bother for now.

**Pull request merged**